service_crategen/src/cargo.rs

```diff
@@ -0,0 +1,52 @@
+use toml;
+
+use std::collections::BTreeMap;
+
+#[derive(Debug, Default, Clone, Deserialize, Serialize)]
+pub struct Manifest {
```

Comment:
I'm surprised this stuff doesn't exist in some already-available crate?  Surely we're not the only ones to build a cargo manifest programmatically.